src/index.js

```diff
@@ -40,6 +40,23 @@ function flat(array, depth = 1) {
   return result;
 };
 
+function flatMap(array, callback) {
+  const mapResult = [];
+  for (let i = 0; i < array.length; ++i) {
+    mapResult.push(callback(array[i], i, array));
+  }
+  let flatResult = [];
+  for (let i = 0; i < mapResult.length; ++i) {
+    if (Array.isArray(mapResult[i])) {
+      flatResult = flatResult.concat((flat(mapResult[i])));
+    } else {
+      flatResult.push(mapResult[i]);
+    }
+  }
+
+  return flatResult;
+};
+
```

Comment:
I much prefer

```function flatMap(array, callback) {
  return flat(map(array, callback);
}```

...but I think that's cheating 😆

Reply:
Nah, that's not cheating at all! You're leveraging your other functions to complete the task, which is DRY 🔥